### PR TITLE
prevent tools & nodes from being dropped

### PR DESCRIPTION
**Pull request by shivajiva101**
admin nodes and tools should not be dropped by a player

### CODE REVIEW

**Comment by Calinou** [comment]
Looks good to me, but there should be a message stating that Map Tools items can't be dropped when the player attempts to drop one.

File: nodes.lua [hunk 1]
```diff
@@ -16,6 +16,10 @@ minetest.register_node(":default:cloud", {
 	drop = "",
 	groups = {unbreakable = 1, not_in_creative_inventory = maptools.creative},
 	sounds = default.node_sound_defaults(),
+	on_drop = function(itemstack, dropper, pos)
+		maptools.msg(dropper)
+		return
```
Comment (SmallJoker) [inline nodes.lua]: Superfluous `return`.

File: init.lua [hunk 2]
```diff
@@ -20,6 +20,11 @@ maptools.intllib = S
 
 local modpath = minetest.get_modpath("maptools")
 
+maptools.msg = function(player)
```
Comment (SmallJoker) [inline init.lua]: Could be changed to `maptools.drop_msg = function(itemstack, player)`, which allows assigning `on_drop` like this:
`on_drop = maptools.drop_msg`

File: nodes.lua (hunk 1 above)
Comment (shivajiva101) [reply nodes.lua]: okay I'll ditch the returns as they are superfluous

**Comment by shivajiva101** [comment]
much more compact, thanks :+1:

**Review by SmallJoker — APPROVED** [review]
Works

**Pull request merged**